ql/src/java/org/apache/hadoop/hive/ql/ddl/table/storage/compact/AlterTableCompactOperation.java

```diff
@@ -62,6 +65,12 @@ public AlterTableCompactOperation(DDLOperationContext context, AlterTableCompact
       throw new HiveException(ErrorMsg.NONACID_COMPACTION_NOT_SUPPORTED, table.getDbName(), table.getTableName());
     }
 
+    if (desc.getPartitionSpec() != null && DDLUtils.isIcebergTable(table) &&
```

Comment:
could we move this logic into AlterTableUtils?
````
isIdentityPartitionSpec(table) {
  return table.getStorageHandler() != null && table.getStorageHandler().getPartitionTransformSpec(table)
      .stream().map(TransformSpec::getTransformType)
      .allMatch(type -> type == TransformSpec.TransformType.IDENTITY);
````
}

Reply:
we should also check if table `hasUndergonePartitionEvolution`, see method in HiveIcebergStorageHandler

Reply:
Done